web/src/lib/components/pages/record.tsx

```diff
@@ -126,6 +146,19 @@ export default class RecordPage extends Component<RecordProps, RecordState> {
     }
   }
 
+  private checkRecording(): string {
```

Comment:
IMO it would be nice to return some kind of status code here (or introduce an enum for the result), so that the checking logic is more separated from the UI (error message)